app/src/main/java/me/rocka/fcitx5test/ui/common/DialogSeekBarPreference.kt

```diff
@@ -0,0 +1,112 @@
+package me.rocka.fcitx5test.ui.common
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.SeekBar
+import androidx.appcompat.app.AlertDialog
+import androidx.preference.Preference
+import androidx.preference.PreferenceManager
+import me.rocka.fcitx5test.R
+import me.rocka.fcitx5test.databinding.DialogSeekBarBinding
+
+/**
+ * Custom preference which represents a seek bar which shows the current value in the summary. The
+ * value can be changed by clicking on the preference, which brings up a dialog which a seek bar.
+ * This implementation also allows for a min / max step value, while being backwards compatible.
+ *
+ * @property defaultValue The default value of this preference.
+ * @property min The minimum value of the seek bar. Must not be greater or equal than [max].
+ * @property max The maximum value of the seek bar. Must not be lesser or equal than [min].
+ * @property step The step in which the seek bar increases per move. If the provided value is less
+ *  than 1, 1 will be used as step.
+ * @property unit The unit to show after the value. Set to an empty string to disable this feature.
+ */
+class DialogSeekBarPreference : Preference {
+    var defaultValue: Int = 0
+    var min: Int = 0
+    var max: Int = 100
+    var step: Int = 1
+    var unit: String = ""
+
+    private val currentValue: Int
+        get() = preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+
+    @Suppress("unused")
```

Comment:
This `@Suppress` is also redundant, actually this is the only constructor we use in `PreferenceScreenFactory`.